src/Tempest/Database/src/IsDatabaseModel.php

```diff
@@ -76,6 +76,11 @@ public static function query(): ModelQueryBuilder
         return new ModelQueryBuilder(self::class);
     }
 
+    public static function getModelInstanceClass(mixed ...$params): string
+    {
+        return self::class;
```

Comment:
I'd go with `static::class`. That way it will still work if people make an abstract model that uses the trait, but make the query from the child model (when they are not overriding `getModelInstanceClass`)

Reply:
I agree in principle but I feel to make that work properly we need an ability to define scopes or default model conditions

For my use-case if I ran `WordPressSource::find($id)` and that ID wasn't for a WordPressSource model I would want it to fail and not find the model rather than proceed anyway with the wrong concrete class and cause unexpected behaviour

Reply:
Aaah yeah, that makes sense! OTOH, in that case you would have overwritten `getModelClass` anyway, wouldn't you? So there would be no reference of `static` or `self` anymore, but rather a reference to the type.

Reply:
Oh I think I get what you're saying - there are 3 scenarios:
1. you have a `final class MyModel` where `static::class` and `self::class` are the same so no biggie
2. my scenario where you customise the class using `getModelInstanceClass()` based on some data condition
3. the calling code says what model class it wants so `static::class` is different to `self::class`

So for scenario 3 you'd have a "base" `MyModel` that may be extended into a "child" `MyModelButDifferent` in certain scenarios and the calling code specifies which one it wants with `MyModel::find()` or `MyModelButDifferent::find()` and gets the one it calls on?

Reply:
Correct! If we don't use `static`, scenario 3 wouldn't work